src/main/java/codeu/controller/BotServlet.java

```diff
@@ -0,0 +1,149 @@
+        if (username == null) {
+            //redirect to login
+            response.sendRedirect("/login");
+            return;
+        }
+        User user = userStore.getUser(username);
+        if (user == null) {
+            // couldn't find author, redirect to loging Page
+            response.sendRedirect("/login");
+            return;
+        }
+
+        Conversation botConversation = conversationStore.getBotConversation(user.getId());
+        if (botConversation == null){
+             botConversation = new Conversation(UUID.randomUUID(), user.getId(), DEFULT_BOT_CONVERSATION_TITLE+username, Instant.now());
+            conversationStore.addConversation(botConversation);
+        }
+        List<Message> messages = messageStore.getMessagesInConversation(botConversation.getId());
+        request.setAttribute("botmessage", messages);
+        request.getRequestDispatcher("/WEB-INF/view/bot.jsp").forward(request, response);
+    }
+
+    /**
+     * This function fires when a user submits the form on the bot chat page. It gets the logged-in
+     * username from the session,  and the chat message from the
+     * submitted form data. It creates a new Message from that data, adds it to the model, and then
+     * redirects back to the chat page.
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+
+        String username = (String) request.getSession().getAttribute("user");
+        if (username == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+        User user = userStore.getUser(username);
+        if (user == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+
+        String userInput = request.getParameter("botmessage");
+
+        Conversation botConversation = conversationStore.getConversationWithTitle(DEFULT_BOT_CONVERSATION_TITLE+username);
+        if (botConversation == null){
+            botConversation = new Conversation(user.getId(), user.getId(), DEFULT_BOT_CONVERSATION_TITLE +username, Instant.now());
+            conversationStore.addConversation(botConversation);
+        }
+        // sanitizes
+        // This removes any HTML from the description content
+        String cleanedMessageContent = Jsoup.clean(userInput, Whitelist.none());
+        Message message = new Message(UUID.randomUUID(),botConversation.getId(), user.getId(),cleanedMessageContent, Instant.now());
+        messageStore.addMessage(message);
+
+        //Call the action matcher class from here to send the message to Action matcher class for every messages
+        //ActionmatcherClass(cleanedMessageContent);
+        List<Message> messages = messageStore.getMessagesInConversation(botConversation.getId());
+        response.sendRedirect("/bot");
+    }
+
+    public UserStore getUserStore() {
```

Comment:
Are you planning to use these get methods? If not, I think that the best would be to erase them.

Reply:
I was using this for unit testing.